tasks/simplification.py

```diff
@@ -0,0 +1,227 @@
+
+    executed = False
+
+    def run(self):
+        session = CurrentSession().get()
+
+        if self.retainfactor == '?':
+            self.retainfactor = DEFAULT_P_RETAIN_FACTOR
+
+        factor = postgis_simplification_factor(self.schema, self.table, self.geomfield, self.retainfactor)
+
+        session.execute('UPDATE "{schema}".{table} '
+                        'SET {geomfield} = ST_MakeValid(ST_SimplifyVW({geomfield}, {factor})); '.format(
+                            schema=self.schema, table=self.table, geomfield=self.geomfield, factor=factor))
+        session.execute("COMMIT")
+
+        self.executed = True
+
+    def output(self):
+        return PostgresTarget(self.schema, self.table)
+
+    def complete(self):
+        if not self.executed:
+            return False
+        return self.output().exists()
+
+
+class MapshaperSimplification():
+    def __init__(self, retainpercentage=DEFAULT_M_RETAIN_PERCENTAGE, skipfailures=SKIPFAILURES_NO):
+        self.retainpercentage = retainpercentage
+        self.skipfailures = skipfailures
+
+
+class PostGISSimplification():
+    def __init__(self, geomfield=DEFAULT_GEOMFIELD, retainfactor=UNKNOWN_RETAIN_FACTOR):
+        self.geomfield = geomfield
+        self.retainfactor = retainfactor
+
+
+class Simplify(Task):
+    schema = Parameter()
+
+    dont_simplify = None
+    '''
+    Array of names of tables in the schema that we don't want to simplify
+    Example: dont_simplify = ['mytable1', 'mytable2']
+    '''
+
+    override_defaults = None
+    '''
+    Array of Dict objects to override default simplification
+    Example: [('mytable', PostGISSimplification(retainfactor=60))]
+    (simplify 'mytable' using a PostGIS simplification with a retain factor of 60)
+    '''
+
+    executed = False
+
+    def requires(self):
+        for _tableid, _tablename in self.find_tables(self.schema):
+            yield self.find_simplification(_tableid, _tablename)
+
+    def run(self):
+        self.executed = True
+
+    def complete(self):
+        return self.executed
+
+    def find_tables(self, schema):
+        session = CurrentSession().get()
+
+        return session.execute("SELECT id, tablename FROM observatory.obs_table "
+                               "WHERE the_geom IS NOT NULL AND id LIKE '{schema}%' "
+                               "{dont_simplify} ".format(
+                                    schema=schema,
+                                    dont_simplify="" if self.dont_simplify is None
+                                                  else "AND id NOT SIMILAR TO '" + self.schema + ".("
+                                                  + "|".join([x for x in self.dont_simplify]) + ")%'"
+                                )).fetchall()
+
+    def find_simplification(self, tableid, tablename):
+        simplification = SimplifyGeometriesMapshaper(OBSERVATORY_SCHEMA, tablename)
+
+        if self.override_defaults is not None:
```

Comment:
You can make `override_defaults` be `[]` by default (instead of `None`) and remove this check.